core/modules/startup/keyboard.js

```diff
@@ -0,0 +1,97 @@
+/*\
+title: $:/core/modules/startup/keyboard.js
+type: application/javascript
+module-type: startup
+
+Keyboard shortcut handling
+
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+// Export name and synchronous status
+exports.name = "global-keyboard";
+exports.platforms = ["browser"];
+exports.after = ["startup"];
+exports.synchronous = true;
+
+exports.startup = function() {
+	if($tw.browser) {
+		// Get existing shortcut-tiddlers
+		var shortcutTiddlers = $tw.wiki.filterTiddlers("[all[shadows+tiddlers]tag[$:/tags/KeyboardShortcut]!has[draft.of]]"),
```

Comment:
First, I think we could get away with calling `$tw.wiki.getTiddlersWithTag()`, which is much faster than compiling and executing a filter.

But more importantly, I think we can improve things here. There are three places where we're duplicating the same logic.